File: go/algorithms/easy/twoStrings.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Two Strings - HackerRank
See: https://www.hackerrank.com/challenges/two-strings/problem
*/
//--------- package ---------
package easy


//--------- functions ---------
// Determines if two strings share a common substring.
func twoStrings(s1 string, s2 string) string {
	var haveCommun string = "NO";
	var hashString map[string]int = make(map[string]int);

	for _, value := range s1 {
		if hashString[string(value)] == 0 {
			hashString[string(value)] = 1;
		}//end if
	}//end for

	for _, value := range s2 {
		if hashString[string(value)] == 1 {
			haveCommun = "YES"; break;
		}//end if
	}//end for

	return haveCommun;
}//twoStrings()

File: java/algorithms/easy/twoStrings.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description Two Strings - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/two-strings/problem}
 */
//---------- Imports ----------
import java.util.Map;
import java.util.HashMap;


//---------- solution ----------
class Result 
{
    /**
     * Determines if two strings share a common substring.
     * 
     * @param {String} s1
     * @param {String} s2
     * @return {String}
     */
    public static String twoStrings(String s1, String s2) 
    {
        String haveCommun = "NO";
        Map<Character, Integer> map = new HashMap<Character, Integer>();

        for (char c : s1.toCharArray())
        { map.put(c, 1); }

        for (char c : s2.toCharArray())
        {
            if (map.containsKey(c))
            { haveCommun = "YES"; break; }
        }//end for

        return(haveCommun);
    }//end twoStrings()
}//end class